integration/spark/src/test/scala/org/apache/carbondata/view/rewrite/MVCreateTestCase.scala

```diff
@@ -414,6 +414,20 @@ class MVCreateTestCase extends QueryTest with BeforeAndAfterAll {
     sql(s"drop materialized view mv32")
   }
 
+  test("test create materialized view having sub-query alias used in projection") {
+    sql("drop materialized view if exists mv_sub")
+    val subQuery = "select empname, sum(result) sum_ut from " +
+                   "(select empname, utilization result from fact_table1) fact_table1 " +
+                   "group by empname"
+    sql("create materialized view mv_sub as " + subQuery)
```

Comment:
can you add a test where MV is created on `select empname, utilization result from fact_table1` and then actual query uses this as subquery?